### PR TITLE
Use the new license API.

### PR DESCRIPTION
**Pull request by teclator**
- It depends on yast/yast-yast2#715

### CODE REVIEW

File: src/modules/ProductLicense.rb [hunk 1]
```diff
@@ -778,6 +785,44 @@ def display_beta(id)
...
+    #
+    # @param product  [Y2Packager::Product] Product instance
+    # @param licenses [Hash<String,String>] License files indexed by language
+    # @return [boolean] true if the license was already accepted; false if it was
+    #   not acceptedd or it is not found.
+    def license_accepted_for?(product, licenses)
+      license_file = licenses["en_US"] || licenses["en"] || licenses[""]
+      content = SCR.Read(path(".target.string"), license_file).to_s
+      if content.empty?
+        log.info "No license was found for #{product.name} in the repository"
```
Comment (jreidinger) [inline src/modules/ProductLicense.rb]: are you sure it is just info? and do you test how it behave for this case?

File: src/modules/ProductLicense.rb [hunk 2]
```diff
@@ -778,6 +785,44 @@ def display_beta(id)
...
+    # Find the product in the given repository
+    #
+    # @param src_id [Integer] Repository ID
+    # @return [Y2Packager::Product,nil] Product or nil if it was not found
+    def repository_product(src_id)
+      product_h = Yast::Pkg.ResolvableProperties("", :product, "").find do |properties|
+        properties["source"] == src_id
+      end
+      if product_h.nil?
+        log.info "No product was found in the repository (#{src_id})" unless product_h
```
Comment (jreidinger) [inline src/modules/ProductLicense.rb]: can this happen at all?

Comment (jreidinger) [reply src/modules/ProductLicense.rb]: and useless `unless`

Comment (imobachgs) [reply src/modules/ProductLicense.rb]: Not for official addons, but I prefer to be on the safe side.

File: src/modules/ProductLicense.rb (hunk 1 above)
Comment (imobachgs) [reply src/modules/ProductLicense.rb]: Unit tests should cover this case. I will raise it to error.

**Pull request merged**